src/metadatad.cc

```diff
@@ -30,8 +30,9 @@ int main(int ac, char** av) {
   if (parse_result) {
     return parse_result;
   }
+  google::HealthChecker health_checker(config);
 
-  google::MetadataAgent server(config);
+  google::MetadataAgent server(config, health_checker);
```

Comment:
There may be another way of doing this, but I'd opt for the way @ACEmilG has right now, so that it's easy to stub out the health checker in unit tests. @igorpeshansky would you suggest just setting the property on the class in the context of a unit test?

Reply:
See my proposed implementation. If we go with that, then the `HealthCheck` object is a no-op unless `SetUnhealthy` is called, so doesn't need to be stubbed at all.